superdesk/storage/migrate.py

```diff
@@ -0,0 +1,40 @@
+
+import codecs
+import superdesk
+
+from flask import current_app as app
+
+
+class MigrateMediaCommand(superdesk.Command):
+    """It will migrate files from Mongo GridFS into Amazon S3 storage."""
+
+    option_list = [
+        superdesk.Option('--limit', '-l', dest='limit', required=False, default=50, type=int)
+    ]
+
+    def run(self, limit):
+        mongo = app.media._storage[0]
+        amazon = app.media._storage[1]
+
+        files = mongo.fs().find()
+        print("starting to migrate {}/{} files".format(limit, files.count()))
+
+        counter = 0
+        for file in files:
+            amazon.put(
```

Comment:
Encapsulating the upload in a `try...except` would be ideal in case of an exception being raised. No need to abort the entire process, and further iterations wouldn't abort on the same resource